### PR TITLE
Issue #16 - Read an Item from an Order

### PR DESCRIPTION
**Pull request by thizume**
- Added a new class called TestOrderItemsServer in test_routes.py (Copied and pasted TestOrderServer class)
- Used an undefined function called "_create_order_items" -> It is supposed to be created in Issue #12

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -253,3 +253,210 @@ def test_query_order_list_by_category(self):
...
+        """Runs before each test"""
+        db.drop_all()  # clean up the last tests
+        db.create_all()  # create new tables
+        self.app = app.test_client()
+
+    def tearDown(self):
+        db.session.remove()
+        db.drop_all()
+
+    def _create_orders(self, count):
```
Comment (smh9854) [inline tests/test_routes.py]: This runs the OrderFactory, not OrderItems factory. 

I'm still trying to figure out how to run subfactories and might need to jump on a call with John to clarify.

File: tests/test_routes.py [hunk 2]
```diff
@@ -253,3 +253,210 @@ def test_query_order_list_by_category(self):
...
+    def test_index(self):
+        """Test the Home Page"""
+        resp = self.app.get("/")
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        data = resp.get_json()
+        self.assertEqual(data["name"], "Order Demo REST API Service")
+
+    def test_get_order_list(self):
+        """Get a list of Orders"""
+        self._create_orders(5)
```
Comment (smh9854) [inline tests/test_routes.py]: Get list of all orders, or get list of all order items?

File: tests/test_routes.py [hunk 3]
```diff
@@ -253,3 +253,210 @@ def test_query_order_list_by_category(self):
...
+        logging.debug(test_order)
+        resp = self.app.post(
+            BASE_URL, json=test_order.serialize(), content_type=CONTENT_TYPE_JSON
+        )
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+        # Make sure location header is set
+        location = resp.headers.get("Location", None)
+        self.assertIsNotNone(location)
+        # Check the data is correct
+        new_order = resp.get_json()
```
Comment (smh9854) [inline tests/test_routes.py]: Check Order table schema for variables. 

Incorrect variables: 
- Name
- Category
- Availability

Comment out or swap the variable names

File: tests/test_routes.py [hunk 4]
```diff
@@ -253,3 +253,210 @@ def test_query_order_list_by_category(self):
...
+    #     test_order = order.serialize()
+    #     test_order["gender"] = "male"    # wrong case
+    #     resp = self.app.post(
+    #         BASE_URL, json=test_order, content_type="application/json"
+    #     )
+    #     self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_update_order(self):
+        """Update an existing Order"""
+        # create a order to update
```
Comment (smh9854) [inline tests/test_routes.py]: This section is for another issue
- Update Order
- Delete Order

**Review by smh9854 — CHANGES_REQUESTED** [review]
Too many code chunks spanning different functions. 

Need to focus on the OrderItem, but we may need to finish the Orders first? 

We should ask John about subfactories

File: tests/test_routes.py (hunk 1 above)
Comment (thizume) [reply tests/test_routes.py]: I deleted this part and write new code based on the sample code.

File: tests/test_routes.py (hunk 2 above)
Comment (thizume) [reply tests/test_routes.py]: I deleted this part and write new code based on the sample code.

File: tests/test_routes.py (hunk 3 above)
Comment (thizume) [reply tests/test_routes.py]: I deleted this part and write new code based on the sample code.

File: tests/test_routes.py (hunk 4 above)
Comment (thizume) [reply tests/test_routes.py]: I deleted this part and write new code based on the sample code.

**Pull request merged**